### PR TITLE
minimap2, py-mappy: change dependency and build on ARM

### PR DESCRIPTION
**Pull request by tkameyama**
I found minibuf2 and py-mappy are build on same source files:
https://github.com/lh3/minimap2/tree/master/python
This PR is changes as follows:
* py-mappy home page and souce file URL is changed to minimap2 URL and add new version.
*  py-mappy dependency is removed from minimap2.
* Both packages can build on ARM, https://github.com/lh3/minimap2/blob/master/README.md#install

### CODE REVIEW

**Comment by tkameyama** [comment]
> 
> 
> What's the difference between the two sets of installations?
* minimap2: execute make and copy minimap2 command to install directory
* py-mappy: standard python package installation.

>  If it's the same source and the same files get installed, we should consider merging both of these packages into a single package.

There seems to be no other package depending on both packages:

    [kameyama@kameyamapc2c spack]$ spack dependents minimap2
    No dependents
    [kameyama@kameyamapc2c spack]$ spack dependents py-mappy
    minimap2

I'm merged both packages to minimap2.

File: var/spack/repos/builtin/packages/minimap2/package.py [hunk 1]
```diff
@@ -17,9 +18,19 @@ class Minimap2(MakefilePackage):
     version('2.10', '52b36f726ec00bfca4a2ffc23036d1a2b5f96f0aae5a92fd826be6680c481c20') 
     version('2.2', '5b68e094f4fa3dfbd9b37d5b654b7715')
 
-    depends_on('py-mappy', type=('build', 'run'))
+    conflicts('target=aarch64', when='@:2.10')
     depends_on('zlib')
+    depends_on('py-cython')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/minimap2/package.py]: This should be `type='build'`

File: var/spack/repos/builtin/packages/minimap2/package.py [hunk 2]
```diff
@@ -17,9 +18,19 @@ class Minimap2(MakefilePackage):
...
 
-    def install(self, spec, prefix):
+    @run_after('install')
+    def install_minimap2(self):
+        make = which('make')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/minimap2/package.py]: You shouldn't need this line. Spack already injects a `make` function into all packages.

**Comment by adamjstewart** [comment]
It looks like #11675 (@xfzhao) also modifies the `py-mappy` package. Did you encounter any issues with Python 3.7?

**Comment by tkameyama** [comment]
> 
> 
> It looks like #11675 (@xfzhao) also modifies the `py-mappy` package. Did you encounter any issues with Python 3.7?

I'm success to new minimap2 version 2.14 with python 3.7.4 and 3.6.8 on x86_64 and aarch64,
and minimap2 2.2 with 3.7.4 on x86_64.

current py-mappy with python 3.7.4 is failed.

Note that current py-mappy package include C code generated by cython 0.26,  and dose not cython.
Merged version is not include this C code, but need cython.

File: var/spack/repos/builtin/packages/minimap2/package.py (hunk 2 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/minimap2/package.py]: I delete `which('make')` and add dependency types.

**Pull request merged**